cmake/Rust.cmake

```diff
@@ -0,0 +1,49 @@
+include(FetchContent)
+
+# Don't let Corrosion's tests interfere with ours.
+set(CORROSION_TESTS OFF CACHE BOOL "" FORCE)
+
+FetchContent_Declare(
+    Corrosion
+    GIT_REPOSITORY https://github.com/ridiculousfish/corrosion
+    GIT_TAG fish
+)
+
+FetchContent_MakeAvailable(Corrosion)
+
+set(fish_rust_target "fish-rust")
+
+set(fish_autocxx_gen_dir "${CMAKE_BINARY_DIR}/fish-autocxx-gen/")
+
+corrosion_import_crate(
+    MANIFEST_PATH "${CMAKE_SOURCE_DIR}/fish-rust/Cargo.toml"
+    FEATURES "fish-ffi-tests"
+)
+
+# We need the build dir because cxx puts our headers in there.
+# Corrosion doesn't expose the build dir, so poke where we shouldn't.
+if (Rust_CARGO_TARGET)
+    set(rust_target_dir "${CMAKE_BINARY_DIR}/cargo/build/${_CORROSION_RUST_CARGO_TARGET}")
```

Comment:
Please note that this is not always the correct path, depending on which Generator is used.  Notably, it's not correct for Visual Studio generators and other Multi-config generators (e.g. Ninja-Multiconfig).

On a related note, have you considered using `cxxbridge` to generate the bindings from the CMake side?
On corrosion master there is a [somewhat experimental function](https://github.com/corrosion-rs/corrosion/blob/25308e54588a6e5b461f998d607ced101913e44c/cmake/Corrosion.cmake#L1456) to add cxx bindings via cxxbridge. It's probably not production ready or anything, but it should be sufficient to showcase how to generate the bindings from the CMake side.

Reply:
> Notably, it's not correct for Visual Studio generators 

We don't support Visual Studio (or really Windows at all - it's either wsl or cygwin).

>other Multi-config generators (e.g. Ninja-Multiconfig)

This seems to be a "ninja but you can do debug and release in one" thing? I think it would be fair to not support that until cmake is retired. The plan is to hopefully do that before the next release.

Unless of course you know how to easily get the correct path?

Reply:
> Unless of course you know how to easily get the correct path?

Ignoring Visual Studio, the path currently is as below. This should be considered an implementation detail, but is also unlikely to change in the near future.
```cmake
if(CMAKE_CONFIGURATION_TYPES)
        set (rust_target_dir  "${CMAKE_BINARY_DIR}/$<CONFIG>/cargo/build/${_CORROSION_RUST_CARGO_TARGET}")
else()
        set (build_dir "${CMAKE_BINARY_DIR}/cargo/build/${_CORROSION_RUST_CARGO_TARGET}")
endif()
```